src/prommis/nanofiltration/tests/test_diafiltration_two_salt.py

```diff
@@ -0,0 +1,223 @@
+def diafiltration_two_salt():
+    """
+    Build a flowsheet with the two-salt diafiltration unit model.
+    """
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.properties = SoluteParameter()
+
+    # Define global parameters
+    m.membrane_thickness = Param(
+        initialize=1e-7,
+        units=units.m,
+        doc="Thickness of membrane (z-direction)",
+    )
+    m.membrane_width = Param(
+        initialize=1,
+        units=units.m,
+        doc="Width of the membrane (x-direction)",
+    )
+    m.membrane_length = Param(
+        initialize=100,
+        units=units.m,
+        doc="Length of the membrane, wound radially",
+    )
+    m.dP = Param(
+        initialize=10,
+        units=units.bar,
+        doc="Pressure applied to membrane",
+    )
+    m.Lp = Param(
+        initialize=0.01,
+        units=units.m / units.h / units.bar,
+        doc="Hydraulic permeability coefficient",
+    )
+    m.feed_flow_volume = Param(
+        initialize=100,
+        units=units.m**3 / units.h,
+        doc="Volumetric flow rate of the feed",
+    )
+    m.feed_conc_mass_lithium = Param(
+        initialize=1.7,
+        units=units.kg / units.m**3,
+        doc="Mass concentration of lithium in the feed",
+    )
+    m.feed_conc_mass_cobalt = Param(
+        initialize=17,
+        units=units.kg / units.m**3,
+        doc="Mass concentration of cobalt in the feed",
+    )
+    m.diafiltrate_flow_volume = Param(
+        initialize=30,
+        units=units.m**3 / units.h,
+        doc="Volumetric flow rate of the diafiltrate",
+    )
+    m.diafiltrate_conc_mass_lithium = Param(
+        initialize=0.1,
+        units=units.kg / units.m**3,
+        doc="Mass concentration of lithium in the diafiltrate",
+    )
+    m.diafiltrate_conc_mass_cobalt = Param(
+        initialize=0.2,
+        units=units.kg / units.m**3,
+        doc="Mass concentration of cobalt in the diafiltrate",
+    )
+
+    m.fs.unit = TwoSaltDiafiltration(
+        property_package=m.fs.properties,
+        membrane_length=m.membrane_length,
+        membrane_width=m.membrane_width,
+        membrane_thickness=m.membrane_thickness,
+        membrane_permeability=m.Lp,
+        applied_pressure=m.dP,
+        feed_flow_volume=m.feed_flow_volume,
+        feed_conc_mass_lithium=m.feed_conc_mass_lithium,
+        feed_conc_mass_cobalt=m.feed_conc_mass_cobalt,
+        diafiltrate_flow_volume=m.diafiltrate_flow_volume,
+        diafiltrate_conc_mass_lithium=m.diafiltrate_conc_mass_lithium,
+        diafiltrate_conc_mass_cobalt=m.diafiltrate_conc_mass_cobalt,
+        NFEx=5,
+        NFEz=5,
+    )
+
+    return m
+
+
+@pytest.mark.unit
+def test_config(diafiltration_two_salt):
+    assert len(diafiltration_two_salt.fs.unit.config) == 17
+
+    assert not diafiltration_two_salt.fs.unit.config.dynamic
+    assert not diafiltration_two_salt.fs.unit.config.has_holdup
+
+    # TODO: add additional relevant assertions
```

Comment:
Check that the config parameters have the correct values - see my earlier comment on building these inside the unit model instead of as config arguments, if you did that you would then add assertions in `test_build` below that each config argument creates a `Param` with the expected value.

Reply:
For the test, I think checking the value is useful, especially if the set value is different than the default value.